lib/Dialect/TTIR/Analysis/TTNNOpInterface.cpp

```diff
@@ -0,0 +1,196 @@
+  assert(producer_ops.size() == num_operands);
+  assert(producer_layouts.size() == num_operands);
+
+  // serialize mlir structures to interface structures
+  std::vector<std::vector<uint32_t>> input_shapes(num_operands);
+  std::vector<std::string> input_data_types(num_operands);
+  std::vector<ttnn::mlir_interface::memory_config_tuple> input_memory_configs(
+      num_operands);
+  std::vector<std::string> input_tensor_layouts(num_operands);
+
+  for (unsigned int input_idx = 0; input_idx < num_operands; input_idx++) {
+    input_shapes[input_idx] =
+        ttnn_wrapper::memref_get_tensor_shape(consumer_layout.getMemref());
+    input_data_types[input_idx] =
+        ttnn_wrapper::memref_get_element_type(consumer_layout.getMemref());
+    input_memory_configs[input_idx] =
+        ttnn_wrapper::layout_get_memory_config(consumer_layout);
+    input_tensor_layouts[input_idx] =
+        ttnn_wrapper::layout_get_tensor_layout(consumer_layout);
+  }
+
+  auto output_shape =
+      ttnn_wrapper::memref_get_tensor_shape(consumer_layout.getMemref());
+  std::string output_data_type =
+      ttnn_wrapper::memref_get_element_type(consumer_layout.getMemref());
+  ttnn::mlir_interface::memory_config_tuple output_memory_config =
+      ttnn_wrapper::layout_get_memory_config(consumer_layout);
+  std::string output_tensor_layout =
+      ttnn_wrapper::layout_get_tensor_layout(consumer_layout);
+
+  bool is_valid = false;
+  if (llvm::isa<MultiplyOp>(consumer_op) || llvm::isa<AddOp>(consumer_op) ||
+      llvm::isa<SubtractOp>(consumer_op)) {
+    is_valid =
+        ttnn::mlir_interface::does_binary_op_support_input_output_constraints(
+            input_shapes[0], input_memory_configs[0], input_data_types[0],
+            input_tensor_layouts[0], input_shapes[1], input_memory_configs[1],
+            input_data_types[1], input_tensor_layouts[1], output_memory_config,
+            output_data_type);
+  } else if (llvm::isa<SoftmaxOp>(consumer_op)) {
+    is_valid =
+        ttnn::mlir_interface::does_softmax_op_support_input_output_constraints(
+            input_shapes[0], input_memory_configs[0], input_data_types[0],
+            input_tensor_layouts[0], output_shape, output_memory_config,
+            output_data_type);
+  } else if (llvm::isa<ReluOp>(consumer_op)) {
+    is_valid =
+        ttnn::mlir_interface::does_unary_op_support_input_output_constraints(
+            "RELU", // todo agree upon mapping to ttnn::mlir_interface
+            input_shapes[0], input_memory_configs[0], input_data_types[0],
+            input_tensor_layouts[0], output_shape, output_memory_config,
+            output_data_type);
+  } else {
+    llvm::outs() << consumer_op->getName() << " missing ttnn interface\n";
+    is_valid = false;
+  }
+
+  return is_valid;
+}
+
+bool is_op_configuration_valid(Operation *consumer_op,
+                               const LayoutAttr &consumer_layout) {
+  // If not operand data is provided, assume all operands have the same layout
```

Comment:
this disclaimer has a non-trivial weight. I suggest to move it to the header so it might be more discoverable